tests/api_test.py

```diff
@@ -383,6 +384,7 @@ def f(args_list):
 
   # Jit and Donate arguments
 
+  @jax.numpy_dtype_promotion('standard')  # Test explicitly exercises implicit dtype promotion.
```

Comment:
Is this really supposed to be testing promotion?

In general I'd rather avoid promotion if it's not the point of the test, and it pretty clearly is not here.

Reply:
Argument donation behavior depends on the dtype of the inputs, and this test explicitly passes inputs with different types to the function. With that in mind, I think standard dtype promotion is required to test this behavior. Would you prefer I cast the inputs manually so we can test the same behavior under strict promotion?